internal/controllers/cluster/cluster_controller.go

```diff
@@ -77,14 +77,16 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	log := ctrl.LoggerFrom(ctx)
+	log = log.WithValues("controller", "cluster_controller")
```

Comment:
```suggestion
	predicateLog := ctrl.LoggerFrom(ctx).WithValues("controller", "cluster")
```

Reply:
I think it would be good to use a different name for the variable, so it's only used exactly for the predicates.

The value for the controller field should be one of:
* value of the Named parameter of ctrl.NewControllerManagedBy (if it exists
* otherwise lower case of the kind, i.e. Cluster => "cluster"

Reply:
I think doing this for all controllers / predicates would be a good improvement.

Then we'll need another round of changes to our predicates. But I think that requires some discussions as it would need breaking changes to the predicate func signatures

Reply:
Sure, Thanks for the review, I will update accordingly.